### ISSUE TITLE
Point in time recovery with an external cluster only works with targetTime

### ISSUE THREAD

**Issue by gbartolini** [cloudnative-pg/cloudnative-pg#233]
Currently, CloudNativePG does not correctly handle point in time recovery from a recovery object store via the external cluster definition (using `source`) when any of these options is specified:

- targetXID
- targetName
- targetLSN
- targetImmediate

The reason is that the operator at the moment only selects the latest available base backup.

**IMPORTANT:** this does not affect the primary method to perform PITR, which is via time stamp. However, if for example you want to recover to a restore point that you previously created and therefore specify `targetName`, you have no way to instruct from which base backup to start the recovery.

As a result, it makes sense to add another option called `backupId` in the `recoveryTarget` section to be used only when `source` is specified. By default `backupId` is empty, meaning that the operator will automatically detect the backup from which to start the recovery in case `targetTime` is specified, or use the latest available. If non empty, the operator will not perform the automated detection algorithm and always use that backup ID as start of the recovery process - entirely leaving the responsibility to the end user.

**Comment by phisco** [comment]
We discussed it earlier and we agreed that we should fix the documentation to reflect what we are currently handling.
The current code should be able (to be checked if that's actually the case), given one of the following parameters, `targetTime`, `targetLSN`, and `targetTLI`, to automatically detect the latest backup w.r.t. that target and restore up to the target starting from that backup, taking into consideration also the `targetTLI` if specified in the first two cases.
However we agreed also that users should be able to restore from a specific backup and then proceed replaying WALs up to the given target, therefore we should add a `backupID` field that should be used to bypass the above heuristic.
Relevant code can be found [here](https://github.com/cloudnative-pg/cloudnative-pg/blob/main/pkg/management/catalog/catalog.go#L80) and the `backupID` could probably be added to the [RecoveryTarget](thttps://github.com/cloudnative-pg/cloudnative-pg/blob/main/pkg/management/catalog/catalog.go#L80) struct.

**Comment by gbartolini** [comment]
Please look at the comments I left in GitHub.

IMHO, we should not disable timestamp recovery when passing a backup Id.

**Comment by gbartolini** [comment]
@mnencia I have made the changes in the patch to reflect the initial requirements. Can you please have a look at the patch before we merge it?

**Comment by gbartolini** [comment]
I suggest the following commit message:

```
Up to now we were not properly supporting point in time recovery when a
transaction ID, a named restore point, or immediate target where requested.
The reason is that in this case we would simply use the last available backup
in the catalog. Effectively, only `targetTime` (most popular) and `targetLSN`
were working.

The fix has been implemented by introducing a new option called `backupID` in
the `recoveryTarget` section.

By default `backupID` is empty, meaning that the operator automatically detects
the backup from which to start the recovery in case `targetTime` or `targetLSN`
are specified. Alternatively, use the latest available backup as fallback.

If `backupID` is provided, the operator will always use that backup ID as the
start of the recovery process - entirely leaving the responsibility to the end
user to correctly pick the base backup for recovery.

**IMPORTANT:** Although this is normally seen as a new feature, we have decided
to consider this as a bug fix and made it eligible for backporting too (the
problem exists in 1.15.x and must be solved).

Closes #233

Signed-off-by: Jonathan Battiato <jonathan.battiato@enterprisedb.com>
Signed-off-by: Jonathan Gonzalez V <jonathan.gonzalez@enterprisedb.com>
Signed-off-by: Philippe Scorsolini <p.scorsolini@gmail.com>
Signed-off-by: John Long <john.long@enterprisedb.com>
Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>

Co-authored-by: Jonathan Gonzalez V <jonathan.gonzalez@enterprisedb.com>
Co-authored-by: Philippe Scorsolini <p.scorsolini@gmail.com>
Co-authored-by: John Long <john.long@enterprisedb.com>
Co-authored-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>
```
